interfaces/builtin/dbus_bind.go

```diff
@@ -0,0 +1,380 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/release"
+)
+
+var dbusBindPermanentSlotAppArmorShared = []byte(`
+# Description: Allow owning a name on DBus public bus
+
+#include <abstractions/###DBUS_BIND_ABSTRACTION###>
+
+# register on DBus
+dbus (send)
+    bus=system
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="{Request,Release}Name"
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+dbus (send)
+    bus=system
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="GetConnectionUnix{ProcessID,User}"
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+`)
+
+var dbusBindPermanentSlotAppArmorIndividual = []byte(`
+# bind to a well-known DBus name: ###DBUS_BIND_NAME###
+dbus (bind)
+    bus=###DBUS_BIND_BUS###
+    name=###DBUS_BIND_NAME###,
+
+# Allow us to talk to dbus-daemon
+dbus (receive)
+    bus=###DBUS_BIND_BUS###
+    path=###DBUS_BIND_PATH###
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+dbus (send)
+    bus=###DBUS_BIND_BUS###
+    path=###DBUS_BIND_PATH###
+    interface=org.freedesktop.DBus.Properties
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+`)
+
+var dbusBindPermanentSlotAppArmorIndividualClassic = []byte(`
+# allow unconfined clients talk to ###DBUS_BIND_NAME### on classic
+dbus (receive)
+    bus=###DBUS_BIND_BUS###
+    path=###DBUS_BIND_PATH###
+    peer=(label=unconfined),
+`)
+
+var dbusBindPermanentSlotSecComp = []byte(`
+# Description: Allow owning a name on DBus public bus
+
+getsockname
+recvmsg
+sendmsg
+sendto
+`)
+
+var dbusBindConnectedSlotAppArmorIndividual = []byte(`
+# Description: Allow DBus consumer to connect to ###DBUS_BIND_NAME###
+
+# Communicate with the well-known named DBus service
```

Comment:
Note that the content interface is constrained to snaps of the same publisher only, or specially whitelisted cases.

That said, perhaps we can build something more generic for dbus, based on the fact we get the name of the interface allowed.

Let's discuss that at the sprint.